src/NHibernate/Hql/Ast/ANTLR/Tree/SelectClause.cs

```diff
@@ -493,10 +493,14 @@ private void InitAliases(List<ISelectExpression> selectExpressions)
 					continue;
 				}
 
-				var node = (IASTNode) e;
-				if (processedElements.Add(fromElement))
+				var isNewFrom = processedElements.Add(fromElement);
+				if (isNewFrom)
 				{
 					combinedFromElements.Add(fromElement);
+				}
+				var node = (IASTNode) e;
+				if (isNewFrom || node.Type == SqlGenerator.DOT)
+				{
```

Comment:
Though "it works", I do not think this is correct. I need to analyze more what does the code with `combinedFromElements`, but it sounds likely to me it should be renamed to something like `combinedEntityFromElements` and fetched only with non-shallow `fromElement`, used to select the whole entity, not just its id.